Return NoOpDeviceGuardImpl in replace of CudaDeviceGuard when device is not available, or cpu-only build

SherlockNoMad opened the pull request:
Summary: 

To support exporting a cuda model on a CPU-only machine under fake tensor mode. 
User commonly need to move sample inputs to the cuda device with .to("cuda:0") or .to("cuda") call. 
This diff supports this.
I expect the following pattern to work
```
with FakeTensorMode(allow_non_fake_inputs=True):
    cuda_module = module.to("cuda:0")
    cuda_sample_inputs = tuple([x.to("cuda:0") for x in sample_inputs])
    with torch.no_grad():
        ep = torch.export.export(cuda_module, cuda_sample_inputs)
```

Test Plan:
CI

Rollback Plan:

ezyang commented:
USE_CUDA macro not allowed in c10 alas

SherlockNoMad commented:
> USE_CUDA macro not allowed in c10 alas

ok, removed use of USE_CUDA. 

in cpu-only build, `device_guard_impl_registry[static_cast<size_t>(type) & 0xFF].load();` would return a nullptr.

This should be sufficient to check if we are in cpu-only build.

zou3519 commented on c10/core/impl/DeviceGuardImplInterface.h:
```diff
@@ -360,6 +360,18 @@ inline const DeviceGuardImplInterface* getDeviceGuardImpl(DeviceType type) {
+      return new NoOpDeviceGuardImpl<DeviceType::CUDA>();
+    }
+  }
+
```
I'm fine with how this looks. Can you add a test?

SherlockNoMad replied:
test added. 

notice that I also change the impl to override the guard registry.

SherlockNoMad commented:
cc @ezyang 

I decided not to override the guard registry when cuda is initialized, coz in cpu-only build, cuda doesn't get init. 
Instead, I am overriding the registry lazily.

ezyang commented:
The main downside is that you now must now perform an (admittedly easily predictable--you can make it better with C10_UNLIKELY) branch every time the device guard impl is fetched, which is reasonably hot. Even when you have a CPU only build, there is some code that triggers cuda init when you attempt to go CUDA codepath right?

That being said, I'm not going to block this PR if you really want to do it this way. Just letting you know!

zou3519 commented:
@ezyang I think we are doing it in the way in this PR because we dont know how else to do it lol.

That being said, it sounds like the alternative solution is to do the following somewhere during initialization, but in a hopefully thread-safe way:
```
tls_fake_device_guard = std::make_unique<NoOpDeviceGuardImpl<DeviceType::CUDA>>();
device_guard_impl_registry[cuda_idx].store(tls_fake_device_guard.get());
```
NB: I pulled these lines from torchdistx, which seems like it ran into the same exact problem and decided to patch the registry once.

SherlockNoMad committed:
Return NoOpDeviceGuardImpl in replace of CudaDeviceGuard when device is not available, or cpu-only build (#160532)

Summary:

To support exporting a cuda model on a CPU-only machine under fake tensor mode. 
User commonly need to move sample inputs to the cuda device with .to("cuda:0") or .to("cuda") call. 
This diff supports this.
I expect the following pattern to work
with FakeTensorMode(allow_non_fake_inputs=True):
    cuda_module = module.to("cuda:0")
    cuda_sample_inputs = tuple([x.to("cuda:0") for x in sample_inputs])
    with torch.no_grad():
        ep = torch.export.export(cuda_module, cuda_sample_inputs)

Test Plan:
CI

Rollback Plan:

Reviewed By: ezyang, henryoier

Differential Revision: D80181887

huydhn commented:
Distributed CPU tests start failing in a weird way after this lands, so I'm trying to revert it to see if trunk recovered.  Here is an example failure 

[GH job link](https://github.com/pytorch/pytorch/actions/runs/17690067221/job/50282463494) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/a956c4ab1cb13079203a8f07eb26218724f54dc8)

What I see if that the second test always failed, but it became ok once it was retried in a new process.  Here is an example failure logs:

```
2025-09-13T05:56:05.3323389Z distributed/tensor/test_dtensor_ops.py::TestDTensorOpsCPU::test_dtensor_op_db___rpow___cpu_float32 PASSED [0.3022s] [  0%]
2025-09-13T05:56:05.3324589Z distributed/tensor/test_dtensor_ops.py::TestDTensorOpsCPU::test_dtensor_op_db___rsub___cpu_float32 Fatal Python error: Segmentation fault
2025-09-13T05:56:05.3325231Z 
2025-09-13T05:56:05.3325449Z Thread 0x00007f882ffff640 (most recent call first):
2025-09-13T05:56:05.3326085Z   File "/opt/conda/envs/py_3.10/lib/python3.10/linecache.py", line 72 in checkcache
2025-09-13T05:56:05.3326856Z   File "/opt/conda/envs/py_3.10/lib/python3.10/traceback.py", line 379 in extract
2025-09-13T05:56:05.3327735Z   File "/opt/conda/envs/py_3.10/lib/python3.10/traceback.py", line 227 in extract_stack
2025-09-13T05:56:05.3328660Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py", line 1309 in __init__
2025-09-13T05:56:05.3329997Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 166 in _propagate_tensor_meta_non_cached
2025-09-13T05:56:05.3331314Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 340 in propagate_op_sharding_non_cached
2025-09-13T05:56:05.3332546Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 46 in __call__
2025-09-13T05:56:05.3333766Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 327 in propagate
2025-09-13T05:56:05.3334890Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_dispatch.py", line 155 in dispatch
2025-09-13T05:56:05.3336027Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_api.py", line 355 in __torch_dispatch__
2025-09-13T05:56:05.3337082Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 1044 in _fn
2025-09-13T05:56:05.3337941Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_compile.py", line 53 in inner
2025-09-13T05:56:05.3338813Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_tensor.py", line 1075 in __rsub__
2025-09-13T05:56:05.3339697Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_tensor.py", line 45 in wrapped
2025-09-13T05:56:05.3340716Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 602 in run_dtensor_crossref
2025-09-13T05:56:05.3341857Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 524 in test
2025-09-13T05:56:05.3342828Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 644 in check_dtensor_func
2025-09-13T05:56:05.3343887Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 531 in run_opinfo_test
2025-09-13T05:56:05.3344975Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 539 in test_dtensor_op_db
2025-09-13T05:56:05.3346077Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2360 in wrapper
2025-09-13T05:56:05.3347205Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 1135 in test_wrapper
2025-09-13T05:56:05.3348395Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 1644 in wrapper
2025-09-13T05:56:05.3349515Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 426 in instantiated_test
2025-09-13T05:56:05.3350655Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3223 in wrapper
2025-09-13T05:56:05.3351825Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1223 in wrapper
2025-09-13T05:56:05.3353074Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1324 in run_test_with_threaded_pg
2025-09-13T05:56:05.3354304Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1309 in _run
2025-09-13T05:56:05.3355191Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 953 in run
2025-09-13T05:56:05.3356010Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
2025-09-13T05:56:05.3356831Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 973 in _bootstrap
2025-09-13T05:56:05.3357298Z 
2025-09-13T05:56:05.3357472Z Thread 0x00007f883e716640 (most recent call first):
2025-09-13T05:56:05.3358083Z   File "/opt/conda/envs/py_3.10/lib/python3.10/linecache.py", line 72 in checkcache
2025-09-13T05:56:05.3358833Z   File "/opt/conda/envs/py_3.10/lib/python3.10/traceback.py", line 379 in extract
2025-09-13T05:56:05.3359523Z   File "/opt/conda/envs/py_3.10/lib/python3.10/traceback.py", line 227 in extract_stack
2025-09-13T05:56:05.3360417Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py", line 1309 in __init__
2025-09-13T05:56:05.3361656Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 166 in _propagate_tensor_meta_non_cached
2025-09-13T05:56:05.3363036Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 340 in propagate_op_sharding_non_cached
2025-09-13T05:56:05.3364268Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 46 in __call__
2025-09-13T05:56:05.3365412Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 327 in propagate
2025-09-13T05:56:05.3366532Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_dispatch.py", line 155 in dispatch
2025-09-13T05:56:05.3367630Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_api.py", line 355 in __torch_dispatch__
2025-09-13T05:56:05.3368714Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 1044 in _fn
2025-09-13T05:56:05.3369060Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_compile.py", line 53 in inner
2025-09-13T05:56:05.3369480Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_tensor.py", line 1075 in __rsub__
2025-09-13T05:56:05.3369992Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_tensor.py", line 45 in wrapped
2025-09-13T05:56:05.3370534Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 602 in run_dtensor_crossref
2025-09-13T05:56:05.3370936Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 524 in test
2025-09-13T05:56:05.3371436Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 644 in check_dtensor_func
2025-09-13T05:56:05.3371887Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 531 in run_opinfo_test
2025-09-13T05:56:05.3372344Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 539 in test_dtensor_op_db
2025-09-13T05:56:05.3372843Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2360 in wrapper
2025-09-13T05:56:05.3373402Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 1135 in test_wrapper
2025-09-13T05:56:05.3373958Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 1644 in wrapper
2025-09-13T05:56:05.3374510Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 426 in instantiated_test
2025-09-13T05:56:05.3375018Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3223 in wrapper
2025-09-13T05:56:05.3375515Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1223 in wrapper
2025-09-13T05:56:05.3376074Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1324 in run_test_with_threaded_pg
2025-09-13T05:56:05.3376715Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1309 in _run
2025-09-13T05:56:05.3376990Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 953 in run
2025-09-13T05:56:05.3377352Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
2025-09-13T05:56:05.3377645Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 973 in _bootstrap
2025-09-13T05:56:05.3377652Z 
2025-09-13T05:56:05.3377885Z Current thread 0x00007f883d314640 (most recent call first):
2025-09-13T05:56:05.3378370Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py", line 1394 in __enter__
2025-09-13T05:56:05.3379042Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 166 in _propagate_tensor_meta_non_cached
2025-09-13T05:56:05.3379639Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 340 in propagate_op_sharding_non_cached
2025-09-13T05:56:05.3380133Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 46 in __call__
2025-09-13T05:56:05.3380675Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 327 in propagate
2025-09-13T05:56:05.3381122Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_dispatch.py", line 155 in dispatch
2025-09-13T05:56:05.3381713Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_api.py", line 355 in __torch_dispatch__
2025-09-13T05:56:05.3382105Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 1044 in _fn
2025-09-13T05:56:05.3382494Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_compile.py", line 53 in inner
2025-09-13T05:56:05.3382857Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_tensor.py", line 1075 in __rsub__
2025-09-13T05:56:05.3383255Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_tensor.py", line 45 in wrapped
2025-09-13T05:56:05.3383817Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 602 in run_dtensor_crossref
2025-09-13T05:56:05.3384220Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 524 in test
2025-09-13T05:56:05.3384721Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 644 in check_dtensor_func
2025-09-13T05:56:05.3385166Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 531 in run_opinfo_test
2025-09-13T05:56:05.3385661Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 539 in test_dtensor_op_db
2025-09-13T05:56:05.3386121Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2360 in wrapper
2025-09-13T05:56:05.3386678Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 1135 in test_wrapper
2025-09-13T05:56:05.3387206Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 1644 in wrapper
2025-09-13T05:56:05.3387747Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 426 in instantiated_test
2025-09-13T05:56:05.3388257Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3223 in wrapper
2025-09-13T05:56:05.3388760Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1223 in wrapper
2025-09-13T05:56:05.3389402Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1324 in run_test_with_threaded_pg
2025-09-13T05:56:05.3389957Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1309 in _run
2025-09-13T05:56:05.3390262Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 953 in run
2025-09-13T05:56:05.3390581Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
2025-09-13T05:56:05.3390879Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 973 in _bootstrap
2025-09-13T05:56:05.3390884Z 
2025-09-13T05:56:05.3391084Z Thread 0x00007f883dd15640 (most recent call first):
2025-09-13T05:56:05.3391405Z   File "/opt/conda/envs/py_3.10/lib/python3.10/linecache.py", line 72 in checkcache
2025-09-13T05:56:05.3391737Z   File "/opt/conda/envs/py_3.10/lib/python3.10/traceback.py", line 379 in extract
2025-09-13T05:56:05.3392036Z   File "/opt/conda/envs/py_3.10/lib/python3.10/traceback.py", line 227 in extract_stack
2025-09-13T05:56:05.3392528Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py", line 1309 in __init__
2025-09-13T05:56:05.3393160Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 166 in _propagate_tensor_meta_non_cached
2025-09-13T05:56:05.3393733Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 340 in propagate_op_sharding_non_cached
2025-09-13T05:56:05.3394313Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 46 in __call__
2025-09-13T05:56:05.3394815Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_sharding_prop.py", line 327 in propagate
2025-09-13T05:56:05.3395333Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_dispatch.py", line 155 in dispatch
2025-09-13T05:56:05.3395821Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/distributed/tensor/_api.py", line 355 in __torch_dispatch__
2025-09-13T05:56:05.3396295Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 1044 in _fn
2025-09-13T05:56:05.3396678Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_compile.py", line 53 in inner
2025-09-13T05:56:05.3397110Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_tensor.py", line 1075 in __rsub__
2025-09-13T05:56:05.3397458Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/_tensor.py", line 45 in wrapped
2025-09-13T05:56:05.3397927Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 602 in run_dtensor_crossref
2025-09-13T05:56:05.3398355Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 524 in test
2025-09-13T05:56:05.3398792Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 644 in check_dtensor_func
2025-09-13T05:56:05.3399329Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 531 in run_opinfo_test
2025-09-13T05:56:05.3399804Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 539 in test_dtensor_op_db
2025-09-13T05:56:05.3400319Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 2360 in wrapper
2025-09-13T05:56:05.3400860Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 1135 in test_wrapper
2025-09-13T05:56:05.3401437Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 1644 in wrapper
2025-09-13T05:56:05.3402046Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 426 in instantiated_test
2025-09-13T05:56:05.3402541Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3223 in wrapper
2025-09-13T05:56:05.3403180Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1223 in wrapper
2025-09-13T05:56:05.3403764Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1324 in run_test_with_threaded_pg
2025-09-13T05:56:05.3404290Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1309 in _run
2025-09-13T05:56:05.3404533Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 953 in run
2025-09-13T05:56:05.3404904Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
2025-09-13T05:56:05.3405260Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 973 in _bootstrap
2025-09-13T05:56:05.3405265Z 
2025-09-13T05:56:05.3405440Z Thread 0x00007f88983da740 (most recent call first):
2025-09-13T05:56:05.3405821Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 1116 in _wait_for_tstate_lock
2025-09-13T05:56:05.3406093Z   File "/opt/conda/envs/py_3.10/lib/python3.10/threading.py", line 1100 in join
2025-09-13T05:56:05.3406805Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1339 in _join_threads
2025-09-13T05:56:05.3407322Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_distributed.py", line 1221 in wrapper
2025-09-13T05:56:05.3407650Z   File "/opt/conda/envs/py_3.10/lib/python3.10/unittest/case.py", line 549 in _callTestMethod
2025-09-13T05:56:05.3407970Z   File "/opt/conda/envs/py_3.10/lib/python3.10/unittest/case.py", line 591 in run
2025-09-13T05:56:05.3408456Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3375 in _run_custom
2025-09-13T05:56:05.3408938Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 3405 in run
2025-09-13T05:56:05.3409450Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_device_type.py", line 517 in run
2025-09-13T05:56:05.3409944Z   File "/opt/conda/envs/py_3.10/lib/python3.10/unittest/case.py", line 650 in __call__
2025-09-13T05:56:05.3410319Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/unittest.py", line 333 in runtest
2025-09-13T05:56:05.3410783Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/runner.py", line 169 in pytest_runtest_call
2025-09-13T05:56:05.3411160Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_callers.py", line 121 in _multicall
2025-09-13T05:56:05.3411509Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_manager.py", line 120 in _hookexec
2025-09-13T05:56:05.3411972Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_hooks.py", line 512 in __call__
2025-09-13T05:56:05.3412370Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/runner.py", line 262 in <lambda>
2025-09-13T05:56:05.3412784Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/runner.py", line 341 in from_call
2025-09-13T05:56:05.3413196Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/runner.py", line 261 in call_runtest_hook
2025-09-13T05:56:05.3413614Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/runner.py", line 222 in call_and_report
2025-09-13T05:56:05.3414045Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/runner.py", line 133 in runtestprotocol
2025-09-13T05:56:05.3414597Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pytest_rerunfailures.py", line 549 in pytest_runtest_protocol
2025-09-13T05:56:05.3415021Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_callers.py", line 121 in _multicall
2025-09-13T05:56:05.3415399Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_manager.py", line 120 in _hookexec
2025-09-13T05:56:05.3415857Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_hooks.py", line 512 in __call__
2025-09-13T05:56:05.3416241Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/main.py", line 348 in pytest_runtestloop
2025-09-13T05:56:05.3416712Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_callers.py", line 121 in _multicall
2025-09-13T05:56:05.3417081Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_manager.py", line 120 in _hookexec
2025-09-13T05:56:05.3417447Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_hooks.py", line 512 in __call__
2025-09-13T05:56:05.3417828Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/main.py", line 323 in _main
2025-09-13T05:56:05.3418193Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/main.py", line 269 in wrap_session
2025-09-13T05:56:05.3418669Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/main.py", line 316 in pytest_cmdline_main
2025-09-13T05:56:05.3419064Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_callers.py", line 121 in _multicall
2025-09-13T05:56:05.3419488Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_manager.py", line 120 in _hookexec
2025-09-13T05:56:05.3419842Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/pluggy/_hooks.py", line 512 in __call__
2025-09-13T05:56:05.3420221Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/_pytest/config/__init__.py", line 166 in main
2025-09-13T05:56:05.3420722Z   File "/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torch/testing/_internal/common_utils.py", line 1298 in run_tests
2025-09-13T05:56:05.3421162Z   File "/var/lib/jenkins/workspace/test/distributed/tensor/test_dtensor_ops.py", line 672 in <module>
2025-09-13T05:56:05.3421167Z 
2025-09-13T05:56:05.3439034Z Extension modules: numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, torch._C, torch._C._dynamo.autograd_compiler, torch._C._dynamo.eval_frame, torch._C._dynamo.guards, torch._C._dynamo.utils, torch._C._fft, torch._C._linalg, torch._C._nested, torch._C._nn, torch._C._sparse, torch._C._special, multidict._multidict, yarl._quoting_c, propcache._helpers_c, aiohttp._http_writer, aiohttp._http_parser, aiohttp._websocket.mask, aiohttp._websocket.reader_c, frozenlist._frozenlist, charset_normalizer.md, requests.packages.charset_normalizer.md, requests.packages.chardet.md, thriftpy2.transport.cybase, thriftpy2.transport.buffered.cybuffered, thriftpy2.transport.framed.cyframed, thriftpy2.transport.memory.cymemory, thriftpy2.transport.sasl.cysasl, thriftpy2.protocol.cybin.cybin, yaml._yaml, numba.core.typeconv._typeconv, numba._helperlib, numba._dynfunc, numba._dispatcher, numba.core.runtime._nrt_python, numba.np.ufunc._internal, scipy._lib._ccallback_c, numba.mviewbuf, psutil._psutil_linux, psutil._psutil_posix, scipy._lib._uarray._uarray, scipy.special._ufuncs_cxx, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg._cythonized_array_utils, scipy.linalg._flinalg, scipy.linalg._solve_toeplitz, scipy.linalg._matfuncs_sqrtm_triu, scipy.linalg.cython_lapack, scipy.linalg.cython_blas, scipy.linalg._matfuncs_expm, scipy.linalg._decomp_update, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, scipy.sparse.linalg._isolve._iterative, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.special._ellip_harm_2, scipy.signal._sigtools, scipy.signal._max_len_seq_inner, scipy.signal._upfirdn_apply, scipy.signal._spline, scipy.optimize._minpack2, scipy.optimize._group_columns, scipy._lib.messagestream, scipy.optimize._trlib._trlib, numpy.linalg.lapack_lite, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._cobyla, scipy.optimize._slsqp, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy.optimize.__nnls, scipy.optimize._highs.cython.src._highs_wrapper, scipy.optimize._highs._highs_wrapper, scipy.optimize._highs.cython.src._highs_constants, scipy.optimize._highs._highs_constants, scipy.linalg._interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.spatial._ckdtree, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._distance_wrap, scipy.spatial._hausdorff, scipy.spatial.transform._rotation, scipy.optimize._direct, scipy.integrate._odepack, scipy.integrate._quadpack, scipy.integrate._vode, scipy.integrate._dop, scipy.integrate._lsoda, scipy.interpolate._fitpack, scipy.interpolate.dfitpack, scipy.interpolate._bspl, scipy.interpolate._ppoly, scipy.interpolate.interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, scipy.signal._sosfilt, scipy.ndimage._nd_image, _ni_label, scipy.ndimage._ni_label, scipy.signal._spectral, scipy.special.cython_special, scipy.stats._stats, scipy.stats.beta_ufunc, scipy.stats._boost.beta_ufunc, scipy.stats.binom_ufunc, scipy.stats._boost.binom_ufunc, scipy.stats.nbinom_ufunc, scipy.stats._boost.nbinom_ufunc, scipy.stats.hypergeom_ufunc, scipy.stats._boost.hypergeom_ufunc, scipy.stats.ncf_ufunc, scipy.stats._boost.ncf_ufunc, scipy.stats.ncx2_ufunc, scipy.stats._boost.ncx2_ufunc, scipy.stats.nct_ufunc, scipy.stats._boost.nct_ufunc, scipy.stats.skewnorm_ufunc, scipy.stats._boost.skewnorm_ufunc, scipy.stats.invgauss_ufunc, scipy.stats._boost.invgauss_ufunc, scipy.stats._biasedurn, scipy.stats._levy_stable.levyst, scipy.stats._stats_pythran, scipy.stats._statlib, scipy.stats._mvn, scipy.stats._sobol, scipy.stats._qmc_cy, scipy.stats._rcont.rcont, scipy.signal._peak_finding_utils (total: 159)
2025-09-13T05:56:05.3439301Z Got exit code -11 (SIGSEGV)
```
